tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,289 @@
+                           c[1].rk, testid, c[1].partition_cnt, 0, -1, &mv);
+
+        test_msgver_verify("first.verify", &mv, TEST_MSGVER_ALL, 0, msgcnt);
+
+        TEST_SAY("== Testing consumer restart ==\n");
+        /* Only c[1] should exhibit rebalance behavior */
+        c[1].max_rebalance_cnt++;
+
+        conf = rd_kafka_conf_dup(rd_kafka_conf(c[1].rk));
+        test_consumer_close(c[1].rk);
+        rd_kafka_destroy(c[1].rk);
+
+        c[1].rk = test_create_handle(RD_KAFKA_CONSUMER, conf);
+        rd_kafka_poll_set_consumer(c[1].rk);
+        test_consumer_subscribe(c[1].rk, topics);
+
+        /* Assign */
+        rebalance_start = test_clock();
+        while(!static_member_wait_rebalance(&c[1], rebalance_start,
+              &c[1].assigned_at, 1000))
+                test_consumer_poll_once(c[0].rk, &mv, 10);
+
+        TEST_SAY("== Testing subscription expansion ==\n");
+        /* 
+         * New topics matching the subscription pattern should cause
+         * group rebalance
+         */
+        c[0].max_rebalance_cnt++;
+        c[1].max_rebalance_cnt++;
+
+        rebalance_start = test_clock();
+        test_create_topic(c->rk, tsprintf("%snew", topic), 1, 1);
+
+        /* Revoke */
+        while(!static_member_wait_rebalance(&c[0], rebalance_start,
+              &c[0].revoked_at, 1000))
+                test_consumer_poll_once(c[1].rk, &mv, 10);
+        static_member_wait_rebalance(&c[1], rebalance_start,
+                                     &c[1].revoked_at, -1);
+
+        /* Assign */
+        while(!static_member_wait_rebalance(&c[0], rebalance_start,
+                                            &c[0].assigned_at, 1000))
+                test_consumer_poll_once(c[1].rk, &mv, 10);
+        static_member_wait_rebalance(&c[1], rebalance_start, 
+                                     &c[1].assigned_at, -1);
+
+
+        TEST_SAY("== Testing consumer Unsubscribe ==\n");
+        /* Unsubscribe should send a LeaveGroupRequest invoking a reblance */
+        c[0].max_rebalance_cnt++;
+        c[1].max_rebalance_cnt++;
+
+        rebalance_start = test_clock();
+        rd_kafka_unsubscribe(c[1].rk);
+
+        /* Revoke c1 */
+        static_member_wait_rebalance(&c[1], rebalance_start, &c[1].revoked_at, -1);
+
+        /* Send JoinGroup */
+        test_consumer_subscribe(c[1].rk, topics);
+
+        /* Revoke c0 */
+        static_member_wait_rebalance(&c[0], rebalance_start, &c[0].revoked_at, -1);
+
+        /* Assign */
+        while(!static_member_wait_rebalance(&c[1], rebalance_start,
+                                            &c[1].assigned_at, 1000))
+                test_consumer_poll_once(c[0].rk, &mv, 10);
+
+        static_member_wait_rebalance(&c[0], rebalance_start,
+                                     &c[0].assigned_at, -1);
+
+        TEST_SAY("== Testing max poll violation ==\n");
+        /* max.poll.interval.ms should still be enforced by the consumer */
```

Comment:
It is usually better to move this into its own test (in the same file), which allows the tests to run in parallel (faster!) and makes this test more focused.

But don't change it, just wanted to point it out for future tests.